backend/src/data/repositories/chat-message/chat-message.repository.ts

```diff
@@ -0,0 +1,72 @@
+import { SortOrder } from '~/common/enums/enums';
+import {
+  ChatGetAllMessagesRequestDto,
+  ChatGetLastMessagesRequestDto,
+  ChatMessageCreateRequestDto,
+  ChatMessageGetAllItemResponseDto,
+} from '~/common/types/types';
+import { ChatMessage as ChatMessageM } from '~/data/models/models';
+
+type Constructor = {
+  ChatMessageModel: typeof ChatMessageM;
+};
+
+class ChatMessage {
+  #ChatMessageModel: typeof ChatMessageM;
+
+  public constructor({ ChatMessageModel }: Constructor) {
+    this.#ChatMessageModel = ChatMessageModel;
+  }
+
+  public getAll({
+    chatId,
+  }: ChatGetAllMessagesRequestDto): Promise<
+    ChatMessageGetAllItemResponseDto[]
+  > {
+    return this.#ChatMessageModel
+      .query()
+      .select()
+      .where({ chatId })
+      .withGraphJoined(
+        '[sender(withoutPassword).[userDetails], receiver(withoutPassword).[userDetails]]',
+      )
+      .castTo<ChatMessageGetAllItemResponseDto[]>()
+      .execute();
+  }
+
+  public create({
+    senderId,
+    receiverId,
+    message,
+    chatId,
+  }: ChatMessageCreateRequestDto): Promise<ChatMessageGetAllItemResponseDto> {
+    return this.#ChatMessageModel
+      .query()
+      .insert({ senderId, receiverId, message, chatId: chatId as string })
+      .withGraphFetched(
+        '[sender(withoutPassword).[userDetails], receiver(withoutPassword).[userDetails]]',
+      )
+      .castTo<ChatMessageGetAllItemResponseDto>()
+      .execute();
+  }
+
+  public getLastMessage({
+    userId,
+    chatOpponentId,
+  }: ChatGetLastMessagesRequestDto): Promise<ChatMessageGetAllItemResponseDto> {
+    return this.#ChatMessageModel
+      .query()
+      .select()
+      .where({ senderId: userId, receiverId: chatOpponentId })
+      .orWhere({ senderId: chatOpponentId, receiverId: userId })
```

Comment:
What about using `chatId` here?

Reply:
This was not my scope, we agreed with Alina that she will enchance this part with `chatId`